test/Elastic.Apm.Tests/ApiTests/ConvenientApiSpanTests.cs

```diff
@@ -22,10 +22,10 @@ public class ConvenientApiSpanTests
 	{
 		private const string ExceptionMessage = "Foo";
 		private const string SpanName = "TestSpan";
-		private const int SpanSleepLength = 450;
```

Comment:
This was changed in https://github.com/elastic/apm-agent-dotnet/issues/94 

The final solution was to enable 10% error (e.g. [here](https://github.com/elastic/apm-agent-dotnet/blob/master/test/Elastic.Apm.Tests/ApiTests/ConvenientApiSpanTests.cs#L530) ). The resolution of the timer on the CI machine seems to be very bad, and I'm not sure it'll work... maybe 450 was an overkill from me.

Reply:
Aye, 45 is still > min resolution of the timer. Maybe we should subtract min resolution of the timer in addition to your `10%` rule in the assert

Reply:
Is there a plain managed API to get the timer resolution? I suspect it's different on different OSs (that's why none of us was able to repro the original problem locally). I was unable to find anything...

Reply:
[Stopwatch.Frequency](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.frequency?view=netframework-4.7.2)

```csharp
if (Stopwatch.IsHighResolution)
{
	Console.WriteLine("Operations timed using the system's high-resolution performance counter.");
}
else
{
	Console.WriteLine("Operations timed using the DateTime class.");
}

long frequency = Stopwatch.Frequency;
Console.WriteLine("  Timer frequency in ticks per second = {0}",frequency);
long nanosecPerTick = (1000L * 1000L * 1000L) / frequency;
Console.WriteLine("  Timer is accurate within {0} nanoseconds",nanosecPerTick);

var dt1 = DateTime.UtcNow;
Thread.Sleep(1);
var dt2 = DateTime.UtcNow;
Console.WriteLine((dt2 - dt1).Milliseconds);
```

For me this outputs 

```
Operations timed using the system's high-resolution performance counter.
  Timer frequency in ticks per second = 10000000
  Timer is accurate within 100 nanoseconds
1
```

I image `Stopwatch.IsHighResolution` will be false on CI

Reply:
Nice! 

What about simply setting these sleeps and waits to exactly `nanosecPerTick` (or..well I guess in millisec, since `Thread.Sleep` and `Task.Delay` expect millisec) and assert that the duration is > `nanosecPerTick`? If we set it to something that is within the frequency of the timer then this assert should be correct and we don't need to add error to it.

Reply:
Update: if the resolution is less than 1 millisec I'd just go with 1ms...